wavelets_pytorch/transform.py

```diff
@@ -1,347 +1,347 @@
-
-        if self.complex_wavelet:
-            # Combine real and imag parts, returns object of shape
-            # [n_batch,n_scales,signal_length] of type np.complex128
-            # cwt = (cwt[:,:,0,:] + cwt[:,:,1,:]*1j).astype(self.output_dtype)
-            # cwt = torch.abs(cwt[:,:,0,:] + cwt[:,:,1,:]*1j)  # causes UserWarnings
-            cwt = (cwt[:,:,0,:]**2 + cwt[:,:,1,:]**2)**0.5
-        else:
-            # Just squeeze the chn_out dimension (=1) to obtain an object of shape
-            # [n_batch,n_scales,signal_length] of type np.float64
-            # cwt = np.squeeze(cwt, 2).astype(self.output_dtype)
-            cwt = cwt.squeeze(2)
-
-        # Squeeze batch dimension if single example
-        # if num_examples == 1:
-        #     cwt = cwt.squeeze(0)
-        return cwt
-
-    @property
-    def dt(self):
-        return self._dt
-
-    @dt.setter
-    def dt(self, value):
-        super(WaveletTransformTorch, self.__class__).dt.fset(self, value)
-        self._extractor.set_filters(self._filters)
-
-    @property
-    def signal_length(self):
-        return self._signal_length
-
-    @signal_length.setter
-    def signal_length(self, value):
-        super(WaveletTransformTorch, self.__class__).signal_length.fset(self, value)
+# MIT License
+#
+# Copyright (c) 2018 Tom Runia
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to conditions.
+#
+# Author: Tom Runia
+# Date Created: 2018-04-15
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import six
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+import scipy.signal
+import scipy.optimize
+
+import torch
+from torch.autograd import Variable
+
+from wavelets_pytorch.wavelets import Morlet
+from wavelets_pytorch.network import TorchFilterBank
+
+##########################################################################################
+
+@six.add_metaclass(ABCMeta)
+class WaveletTransformBase(object):
+    """
+
+    Base class for the Continuous Wavelet Transform as described in:
+        "Torrence & Combo, A Practical Guide to Wavelet Analysis (BAMS, 1998)"
+
+    This class is a abstract super class for child classes:
+        WaveletTransform      => implements CWT in SciPy
+        WaveletTransformTorch => implements CWT in PyTorch
+
+    For a more detailed explanation of the parameters, the original code serves as reference:
+        https://github.com/aaren/wavelets/blob/master/wavelets/transform.py#L145
+
+    """
+
+    def __init__(self, dt=1.0, dj=0.125, wavelet=Morlet, unbias=False):

```

Comment:
I would set default value to None in order not to change all my code where this class is already used. And then self.wavelet = Morlet() if wavelet is None else wavelet

Reply:
Done